### ISSUE TITLE
spawnProcess doc glitch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2852]
|<img alt="alanfranzoni's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| alanfranzoni reported|
|-|-|
|Trac ID|trac#2852|
|Type|defect|
|Created|2007-10-17 15:00:04Z|

the current spawnProcess doc, in the childFDs parameter description, reads:

"""
the child will be able to write to that file descriptor and the parent will receive read notification via the IProcessTransport.childDataReceived callback. This is useful for the child's stdout and stderr.
"""

IProcessTransport should be replaced by twisted.internet.protocol.ProcessProtocol, I suppose.

Attachments:

* [interfaces.py_patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/16d/16d2c422094075b5fe527d23a59a19e524675cbe/3ae427b51c429b8e01bbfda37e0ffc30d4b9fc95.py_patch) (2271 bytes) - added by bigdog on 2008-02-04 22:30:45Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2852 2852
type__defect defect
reporter__alanfranzoni alanfranzoni
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__docs docs
time__1192633204000000 1192633204000000
changetime__1203621521000000 1203621521000000
version__None None
owner__bigdog bigdog

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

There should be an IProcessProtocol and it should refer to that.

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bigdog set owner to bigdog<br>bigdog set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|bigdog commented|
|-|-|

3020 will address this by adding IProcessProtocol

**Comment by twisted-trac** [comment]
|<img alt="bigdog's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bigdog set status to `closed`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [22679]) Merge process-interface-3020

Author: bigdog, exarkun
Reviewer: dreid
Fixes [#3020](https://github.com/twisted/twisted/issues/3020)
Refs [#2852](https://github.com/twisted/twisted/issues/2852)

Define an interface for process protocols, twisted.internet.interfaces.IProcessProtocol.
